src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -33,14 +34,20 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        String email = authentication.getName();
+        User user = userService.findByEmail(email).orElseThrow(
+                () -> new RuntimeException("Invalid email"));
```

Comment:
```suggestion
                () -> new RuntimeException("Can't find user by email: "  + email));
```
the same for other similar cases